ts-fold-indicators.el

```diff
@@ -106,7 +106,7 @@
 
 (defun ts-fold-indicators--enable ()
   "Enable `ts-fold-indicators' mode."
-  (if (ts-fold-mode 1)  ; Enable `ts-fold-mode' automatically
+  (if (or ts-fold-mode (ts-fold-mode 1))  ; Enable `ts-fold-mode' automatically
```

Comment:
In my initial revision, there was a circular dependency here of ts-fold triggering ts-fold-indicators triggering ts-fold triggering... So I added this extra check to prevent that. I don't think there's a reason to re-activate ts-fold mode again if it's already on but let me know if I'm missing something.

Reply:
I agree, the design decision here is a bit odd. I can't remember exactly why I added this line, but the idea is to make `ts-fold-indicators-mode` rely on `ts-fold-mode`. And `ts-fold-mode` should rely on `tree-sitter-mode`, so the overall design is snappier.

But the current implementation of `ts-fold` and these plugins (indicator, summary, etc) work on it's own. I haven't made up my mind yet, and I couldn't figure out the best UX for users.

Reply:
One more-general question I have is why is indicators its own mode instead of a setting to be toggled similar to how summaries are done?

Reply:
`ts-summary` only happens in the folding time; unlike `ts-indicators`, you will always need to inspect the buffer's state. Here are a few thoughts:

1. `ts-fold` and `ts-summary` is triggered by keys and executed once.
2. `ts-indicators` will need to listen to the whole buffer to keep indicators up to date
3. That's the odd part since `ts-fold` does not work like `ts-indicators`, so there is no reason to let `ts-indicaotrs` rely on `ts-fold`. `ts-fold-mode` only try to work with `evil` mode.

Maybe we should make indicators mode live on its own since they operate differently. 😅

Reply:
Hm, an independent indicators package would be really interesting! But finding the right abstraction layer could be quite tricky. Anyways, there's no rush for that.

Reply:
Yeah, let's not worry about that part now! ;)